personcreator.rb

```diff
@@ -0,0 +1,39 @@
+class PersonCreator
+  def initialize(library)
+    @library = library
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    choice = gets.chomp
+    if choice == '1'
+      create_student_from_input
+    elsif choice == '2'
+      create_teacher_from_input
+    else
+      puts 'Invalid choice'
+    end
+  end
+
+  def create_student_from_input
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.downcase == 'y'
+    @library.create_student(age, name, permission)
```

Comment:
- [ ]  Everything as regards create_teacher, create_student and co should be in the create_person, since they are persons and it would better separate and simplify things